homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,413 @@
+            await self._controller.set_horizontal_vane(self._deviceid, "swing")
+        elif swing_mode == SWING_HORIZONTAL:
+            await self._controller.set_vertical_vane(self._deviceid, "manual3")
+            await self._controller.set_horizontal_vane(self._deviceid, "swing")
+        elif swing_mode == SWING_VERTICAL:
+            await self._controller.set_vertical_vane(self._deviceid, "swing")
+            await self._controller.set_horizontal_vane(self._deviceid, "manual3")
+
+    async def async_update(self):
+        """Copy values from controller dictionary to climate device."""
+        if self._controller.is_disconnected:
+            if self._connected:
+                self._connected = False
+                _LOGGER.error(
+                    "Connection to %s API was lost. Attempting to reconnect...",
+                    self._devicetype,
+                )
+            await self._controller.connect()
+            self._connection_retries += 1
+        else:
+            if not self._connected:
+                self._connected = True
+                _LOGGER.debug("Restored connection to %s API.", self._devicetype)
+            self._connection_retries = 0
+
+        self._current_temp = self._controller.get_temperature(self._deviceid)
+        self._fan_speed = self._controller.get_fan_speed(self._deviceid)
+        self._power = self._controller.is_on(self._deviceid)
+        self._min_temp = self._controller.get_min_setpoint(self._deviceid)
+        self._max_temp = self._controller.get_max_setpoint(self._deviceid)
+        self._rssi = self._controller.get_rssi(self._deviceid)
+        self._run_hours = self._controller.get_run_hours(self._deviceid)
+        self._target_temp = self._controller.get_setpoint(self._deviceid)
+        mode = self._controller.get_mode(self._deviceid)
+
+        # Operation mode
+        self._hvac_mode = MAP_OPERATION_MODE.get(mode, STATE_UNKNOWN)
+
+        # Swing mode
+        # Climate module only supports one swing setting.
+        if self._has_swing_control:
+            self._vvane = self._controller.get_vertical_swing(self._deviceid)
+            self._hvane = self._controller.get_horizontal_swing(self._deviceid)
+
+            if self._vvane == "swing" and self._hvane == "swing":
+                self._swing = SWING_BOTH
+            elif self._vvane == "swing":
+                self._swing = SWING_VERTICAL
+            elif self._hvane == "swing":
+                self._swing = SWING_HORIZONTAL
+            else:
+                self._swing = SWING_OFF
+
+    async def async_will_remove_from_hass(self):
+        """Shutdown the controller when the device is being removed."""
+        await self._controller.stop()
+
+    @property
+    def icon(self):
+        """Return the icon for the current state."""
+        icon = None
+        if self._power:
+            icon = MAP_STATE_ICONS.get(self._hvac_mode)
+        return icon
+
+    def update_callback(self):
```

Comment:
Will this callback be called from within the event loop? If so we should rename it `async_update_calback` and use the async api inside.